server.js

```diff
@@ -27,9 +19,84 @@ const app = express()
 app.use(cors())
 app.use(bodyParser.json())
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+// set up models (Schema) (define what type of data)
+// you could put all models in a models-folder also.
+const NetflixTitle = new mongoose.model('NetflixTitle', {
+  show_id: Number,
+  title: String,
+  director: String,
+  cast: String,
+  country: String,
+  date_added: String,
+  release_year: Number,
+  rating: String,
+  duration: String,
+  listed_in: String,
+  description: String,
+  type: String
+})
+
+//Populate the database
+if (process.env.RESET_DATABASE) { // environment variable used to reset the database
+  const seedDatabase = async () => {
+    await NetflixTitle.deleteMany(); //main rule: clean database before populating database.
+
+    netflixData.forEach(item => {
+      const newNetflixTitle = new NetflixTitle(item); // item is a single object from the json array.
+      newNetflixTitle.save();
+    })
+  }
+  seedDatabase();
+}
+
+//ENDPOINT TO GET ALL TITLES: '/netflix-titles'
+//QUERY-PARAM TO GET ONE TITLE BY TITLE-NAME: /netflix-titles?title=American 
+//QUERY-PARAM TO GET DIRECTOR: /netflix-titles?director=nate
+//Query-PARAM TO GET ACTOR: /netflix-titles?actor=niro
+//Query-PARAM TO GET TITLES BY TYPE: /netflix-titles?type=tv
+//Query-PARAM TO GET TITLES FROM COUNTRY: /netflix-titles?country=germany
+//EXAMPLE TO FIND TITLE(S) BY DIRECTOR & ACTOR /netflix-titles?actor=niro&director=scorsese
+app.get('/netflix-titles', async (req, res) => {
+  const { title, director, actor, type, country } = req.query;
+  const allTitles = await NetflixTitle.find({
+    title: new RegExp(title, 'i'), //regexp: 'i' is 'ignore case'
+    director: new RegExp(director, 'i'),
+    cast: new RegExp(actor, 'i'),
+    type: new RegExp(type, 'i'),
+    country: new RegExp(country, 'i')
```

Comment:
Cool with the RegExp! 

You have choose some of the keys in the database and excluded others, I am not sure if this was to try it out maybe? Consider doing a map to include all the keys 😄